README.md

```diff
@@ -1 +1,164 @@
+the community.
+
+### Related technology
+
+- Python 3.8
+- Amazon Web Services (AWS)
+
+### Assumptions
+
+- You have basic-good knowledge in python programming.
+- You have basic-good knowledge in AWS and CloudFormation.
+- You have basic-good knowledge in OpenSearch.
+
+### Useful sources
+
+- Read more about Cloud Formation: https://docs.aws.amazon.com/cloudformation/index.html
+- Read more about Opensearch: https://docs.aws.amazon.com/opensearch-service/latest/developerguide/remote-reindex.html
+
+### Install
+
+Use the package manager pip to install this package. This project is not in the PyPi
+repository yet. Install directly from source or PyPI.
+
+```bash
+pip install .
+```
+
+Or
+
+```bash
+pip install b-cfn-opensearch-index
+```
+
+### Usage & Examples
+
+Usage of this AWS CloudFormation custom resource is quite simple. Initialize it within any valid CDK scope giving a unique name, OpenSearch domain endpoint, index name prefix, and settings/mappings of the index.
+
+Bellow an example of simple index initialization for indexing of blog posts at OpenSearch.
+
+```python
+from aws_cdk.core import Stack
+
+from b_cfn_opensearch_index.resource import OpensearchIndex
+
+
+class ExampleOpensearchIndexStack(Stack):
+    def __init__(self, scope: Stack, opensearch_domain_endpoint: str) -> None:
+        super().__init__(scope=scope, id='ExampleOpensearchIndexStack')
+
+        # Define index settings.
+        index_settings = {
+             'index.refresh_interval': '2s',
+        }
+
+        # Define index mappings.
+        index_mappings = {
+            'id': {
+                'type': 'keyword'
+            },
+            'author': {
+                'type': 'keyword'
+            },
+            'title': {
+                'type': 'keyword'
+            },
+            'content': {
+                'type': 'text'
+            },
+            'timestamp': {
+                'type': 'date',
+                'format': 'yyyy-MM-dd HH:mm:ss||epoch_millis'
+            },
+        }
+
+        # Initialize AWS CloudFormation custom resource using given OpenSearch
+        # domain endpoint with defined settings and mappings of the index.
+        self.index = OpensearchIndex(
+            scope=self,
+            name='OpensearchIndex',
+            opensearch_domain_endpoint=opensearch_domain_endpoint,
+            index_prefix='example-index-prefix',
+            index_settings=index_settings
+            index_mapping_settings=index_mappings
+        )
+
+    @property
+    def index_name(self) -> str:
+        """
+        Retrieve OpenSearch index name.
+
+        :return: String representation of index name.
+        """
+
+        return self.index.index_name
+
+```
+
+In the above example, we created two dictionaries holding settings and mappings of the index. More information about settings and mappings of the index can be found here https://opensearch.org/docs/latest/opensearch/rest-api/index-apis/create-index/.
```

Comment:
In the example above, we created an OpenSearch index with two dictionaries for settings and mappings of the index.